web/annotation_layer_builder.js

```diff
@@ -129,6 +129,27 @@ class AnnotationLayerBuilder {
       AnnotationLayer.render(parameters);
       this.l10n.translate(this.div);
     }
+
+    const { width, height, rotation } = viewport;
+    const { style } = this.div;
+
+    if (rotation === 0 || rotation === 180) {
+      style.width = `${width}px`;
+      style.height = `${height}px`;
+    } else {
+      style.width = `${height}px`;
+      style.height = `${width}px`;
+    }
+
+    const { classList } = this.div;
+    for (const angle of [90, 180, 270]) {
+      const rotate = `rotate${angle}`;
+      if (angle === rotation) {
+        classList.add(rotate);
+      } else {
+        classList.remove(rotate);
+      }
+    }
```

Comment:
*Possibly a stupid idea, but:* Could we perhaps simplify this a little bit, by using an attribute instead here?
```javascript
this.div.setAttribute("data-annot-rotation", rotation);
```
And the CSS file would then look something like:
```CSS
[data-annot-rotation="90"] {
  transform: rotate(90deg) translateY(-100%);
}
[data-annot-rotation="180"] {
  transform: rotate(180deg) translate(-100%, -100%);
}
[data-annot-rotation="270"] {
  transform: rotate(270deg) translateX(-100%);
}
```

Please note that the above is untested, but I wanted to ask nonetheless.

Reply:
Good idea and it works :)